services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusControllerITest.kt

```diff
@@ -687,6 +690,128 @@ class HakemusControllerITest(@Autowired override val mockMvc: MockMvc) : Control
         }
     }
 
+    @Nested
+    inner class OperationalCondition {
+        private val url = "/hakemukset/$id/toiminnallinen-kunto"
+        private val date = LocalDate.of(2024, 8, 8)
+        private val request = DateReportRequest(date)
+
+        @Test
+        @WithAnonymousUser
+        fun `returns 401 when unknown user`() {
+            post(url, request).andExpect(status().isUnauthorized)
+
+            verifySequence { hakemusService wasNot Called }
+        }
+
+        @Test
+        fun `returns 400 when no request body`() {
+            post(url).andExpect(status().isBadRequest)
+
+            verifySequence { hakemusService wasNot Called }
+        }
+
+        @Test
+        fun `returns 200 when date is sent successfully`() {
+            every {
+                authorizer.authorizeHakemusId(id, PermissionCode.EDIT_APPLICATIONS.name)
+            } returns true
+            justRun { hakemusService.operationalCondition(id, date) }
+
+            post(url, request).andExpect(status().isOk).andExpect(content().string(""))
+
+            verifySequence {
+                authorizer.authorizeHakemusId(id, PermissionCode.EDIT_APPLICATIONS.name)
+                hakemusService.operationalCondition(id, date)
+            }
+        }
+
+        @Test
+        fun `returns 404 when the application is not found`() {
+            every {
+                authorizer.authorizeHakemusId(id, PermissionCode.EDIT_APPLICATIONS.name)
+            } throws HakemusNotFoundException(id)
+
+            post(url, request)
+                .andExpect(status().isNotFound)
+                .andExpect(hankeError(HankeError.HAI2001))
+
+            verifySequence {
+                authorizer.authorizeHakemusId(id, PermissionCode.EDIT_APPLICATIONS.name)
+            }
+        }
+
+        private fun RuntimeException.whenThrown(f: () -> Unit) {
```

Comment:
Would it be better if this extension function was last and not here in the middle of test methods?

Reply:
Yes, thank you. I moved it to behind the test methods.